### PR TITLE
Fix sqlsrv

### LINKED ISSUE #418: libmsodbcsql-17.10.so.1.1 not found
On the Bref 2 sqlsrv image:

```
An exception occurred in the driver: SQLSTATE[01000]: [unixODBC][Driver Manager]Can't open lib '/opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.10.so.1.1' : file not found
```

### PR DESCRIPTION
**Pull request by ddeboer**
Fix #418

### CODE REVIEW

File: layers/sqlsrv/test.php [hunk 1]
```diff
@@ -17,7 +17,7 @@
 // This test attempts a connection, forcing MS odbc libraries to be loaded
 if (!sqlsrv_connect('localhost', ['LoginTimeout' => 1])) {
     foreach (sqlsrv_errors() as $error) {
-        if ($error['SQLSTATE'] == '01000') {
+        if ($error['SQLSTATE'] == '01000' || $error['SQLSTATE'] == 'IM002') {
```
Comment (ddeboer) [inline layers/sqlsrv/test.php:20]: This makes the build fail, as mentioned in https://github.com/brefphp/extra-php-extensions/issues/418#issuecomment-1528873718. It’s a start.

**Comment by ddeboer** [comment]
Turns out work was already started in #474, so closing this one.

**Comment by ddeboer** [comment]
Re-opening because #474 seems stalled, so continuing work here.

File: layers/sqlsrv/test.php [hunk 2]
```diff
@@ -1,7 +1,5 @@
 <?php
 
-putenv('ODBCSYSINI=/opt/microsoft/conf/');
```
Comment (ddeboer) [inline layers/sqlsrv/test.php:3]: As mentioned in https://github.com/brefphp/extra-php-extensions/pull/474#issuecomment-1690764705, `putenv` doesn’t work.

**Comment by ddeboer** [comment]
@mnapoli Would be great to get this merged to generate progress on this issue.

**Comment by buddhaCode** [comment]
Do you need anyone to test this?

**Comment by ddeboer** [comment]
@buddhaCode While I fully expect this to work (famous last words), if you have some time, please go ahead and test it.

**Review by buddhaCode — APPROVED** [review]

**Comment by ddeboer** [comment]
Thanks for validating the changes @buddhaCode.

@mnapoli @Nyholm This issue is preventing some of us from upgrading to Bref v2, so would be great if you could review/merge this.

**Review by Nyholm — APPROVED** [review]
Thank you for the work and for the reviews.

**Pull request merged**